src/Generator/ShopzillaDE.php

```diff
@@ -41,115 +58,235 @@ public function __construct(ArrayHelper $arrayHelper)
-            {
-                if($idlVariation instanceof Record)
-                {
-                    $this->idlVariations[$idlVariation->variationBase->id] = [
-                        'itemBase.id' => $idlVariation->itemBase->id,
-                        'variationBase.id' => $idlVariation->variationBase->id,
-                        'variationRetailPrice.price' => $idlVariation->variationRetailPrice->price,
-                    ];
-                }
-            }
+            $data = [
+                'Kategorie'         => $this->elasticExportHelper->getCategory((int)$variation['data']['defaultCategories'][0]['id'], $settings->get('lang'), $settings->get('plentyId')),
+                'Hersteller'        => $this->elasticExportHelper->getExternalManufacturerName((int)$variation['data']['item']['manufacturer']['id']),
+                'Bezeichnung'       => $this->elasticExportHelper->getMutatedName($variation, $settings, 256),
+                'Beschreibung'      => $this->elasticExportHelper->getMutatedDescription($variation, $settings, 256),
+                'Artikel-URL'       => $this->elasticExportHelper->getMutatedUrl($variation, $settings, true, false),
+                'Bild-URL'          => $this->elasticExportHelper->getMainImage($variation, $settings),
+                'SKU'               => (string)$this->elasticExportHelper->generateSku($variation['id'], (float)$settings->get('referrerId'), 0, (string)$variation['data']['skus'][0]['sku']),
+                'Bestand'           => 'Auf Lager',
+                'Versandgewicht'    => $variation['data']['variation']['weightG'],
+                'Zustand'           => 'Neu',
+                'Versandkosten'     => $this->getShippingCost($variation),
+                'Gebot'             => '',
+                'Werbetext'         => '2',
+                'EAN'               => $this->elasticExportHelper->getBarcodeByType($variation, $settings->get('barcode')),
+                'Preis'             => $priceList['price'],
+                'Grundpreis'        => $this->elasticExportPriceHelper->getBasePrice($variation, (float)$priceList['price'], $settings->get('lang')),
+            ];
+
+            $this->addCSVContent(array_values($data));
+        }
+        else
+        {
+            $this->getLogger(__METHOD__)->info('ElasticExportShopzillaDE::logs.variationNotPartOfExportPrice', [
+                'VariationId' => $variation['id']
+            ]);
+        }
+    }
+
+    /**
+     * Get the shipping cost.
+     *
+     * @param array $variation
+     * @return string
+     */
+    private function getShippingCost($variation):string
+    {
+        $shippingCost = null;
+        if(isset($this->shippingCostCache) && array_key_exists($variation['data']['item']['id'], $this->shippingCostCache))
+        {
+            $shippingCost = $this->shippingCostCache[$variation['data']['item']['id']];
+        }
+
+        if(!is_null($shippingCost) && $shippingCost != '0.00')
```

Comment:
What happens, if you have another pattern for the shipping cost? Each country has another standard for splitting decimal positions. In Germany, the people don't use dot, they use comma. In UK they use the dot instead of a comma. Pay attention to those details!